modules/dcache/src/main/java/diskCacheV111/poolManager/CostModuleV1.java

```diff
@@ -83,16 +90,61 @@ public ImmutableMap<String, String> getTagMap() {
         public PoolInfo getPoolInfo() {
             return new PoolInfo(_address, _info, _tagMap);
         }
+
+        public long getSerialId() {
+            return _serialId;
+        }
+
+        public int getTrustScore() {
+            return _trustScore;
+        }
+
+        public boolean getEnabledStatus() {
+            return _enabled;
+        }
+    }
+    public boolean getPoolStatus (String poolName) {
+        return _hash.get(poolName).getEnabledStatus();
+
     }
 
     public synchronized void messageArrived(CellMessage envelope, PoolManagerPoolUpMessage msg) {
+        int tsIncrease = 15;
+        int tsDecrease = 2;
+        int tsThreshold = 35;
+        int tsCeiling = 150;
+        long msgSerialId = msg.getSerialId();
+        int nextTrustScore = 0;
+        boolean nextEnabledStatus = true;
+
         CellAddressCore poolAddress = envelope.getSourceAddress();
         String poolName = msg.getPoolName();
         PoolV2Mode poolMode = msg.getPoolMode();
         PoolCostInfo newInfo = msg.getPoolCostInfo();
         Entry poolEntry = _hash.get(poolName);
         boolean isNewPool = poolEntry == null;
 
+        if (!isNewPool) { // Only check for reboots if the pool is not new
+            int lastTrustScore = poolEntry.getTrustScore();
+            long lastSerailId = poolEntry.getSerialId();
+
+            if (msgSerialId == lastSerailId) { // Pool has rebooted
+                nextTrustScore = lastTrustScore/tsDecrease;
+                if (nextTrustScore < tsThreshold && !poolEntry.getEnabledStatus()) { // Pool was disabled, should now be re-ENABLED
+                    LOGGER.error("Pool {} WOULD now be re-ENABLED due to low trust score, BUT IS NOT", poolName);
```

Comment:
the message is confusing....

Reply:
message changed in [a5bd221](https://github.com/dCache/dcache/pull/7532/commits/a5bd2215d433efbb83ec7298f1aef70ced2cd968)